public/src/components/epicTests/epicTestsForm.tsx

```diff
@@ -154,16 +156,40 @@ class EpicTestsForm extends React.Component<EpicTestFormProps, EpicTestsFormStat
   fetchStateFromServer = (): void => {
     fetchFrontendSettings(FrontendSettingsType.epicTests)
       .then(serverData => {
+        const editMode = serverData.status.email === serverData.userEmail;
+
+        this.updateWarningTimeout(editMode);
+
         this.setState({
           ...serverData.value,
           previousStateFromServer: serverData,
           lockStatus: serverData.status,
-          editMode: serverData.status.email === serverData.userEmail,
+          editMode: editMode,
           modifiedTests: {}
         });
       });
   };
 
+  // Maintains an alert if tool is left open for edit for 20 minutes
+  updateWarningTimeout = (editMode: boolean): void => {
+    if (editMode) {
```

Comment:
I would add something like `if (!window) { return }` to ensure the window api is available - sometimes an issue in react

Reply:
that's odd. What would happen if it went ahead and called window.setTimeout, just a console error?

Reply:
discussed IRL and decided that because this logic happens after a `fetch` request, it's likely safe